spec/fixtures/api/utility-process/mocha.js

```diff
@@ -0,0 +1,58 @@
+const path = require('node:path');
```

Comment:
Copying our mocha setup into here is brittle and will result in confusing drift between this and the main setup file. Can we find another way to run these tests in both utility and main process? Perhaps draw inspiration from e.g. the `itremote` helper in spec/lib/spec-helpers.ts?

I'm thinking something like,

```js
describe('net module', () => {
  for (const test of [it, itutility]) {
    test('test name', async () => {
      // ...
    })
  }
})
```

This would keep all the tests in the main runner.

Reply:
In order to be forked and run in the utility process I believe the tests need to be in a file. What about factoring out the common pieces between this and `spec/index.js`? That would prevent drift

Reply:
See `itremote`, it works by sending code to the other process to be dynamically evaluated.

Reply:
I’m not sure how having multiple nested reporters running works for eg reporting test status.

Reply:
> I’m not sure how having multiple nested reporters running works for eg reporting test status.

The main runner sees the nested reporting as a single test.

> See itremote, it works by sending code to the other process to be dynamically evaluated.

Ok I'll take a pass at adding something like `itremote` instead of `contents.executeJavaScript` it can use a `child.postMessage` to a utility process (forked with the common net spec dependencies/setup) and then `eval`ing the code from the message. I'll take a pass at that!

Reply:
Ok took at pass at this in 370e3c6 happy to get feedback! 🙏

Reply:
Resolving since this is done